pkgs/development/compilers/eunomia-bpf/eunomia-cc/default.nix

```
@@ -0,0 +1,128 @@
+{ lib
+, stdenv
+, makeWrapper
+, fetchFromGitHub
+, rustPlatform
+, cmake
+, pkg-config
+, llvmPackages_16
+, cargo
+, rustc
+, elfutils
+, zlib
+, ...
+}:
+let
+  bpftool =
+    stdenv.mkDerivation {
+      pname = "bpftool";
+      version = "eunomia-edition-20230311";
+      src = fetchFromGitHub {
+        owner = "eunomia-bpf";
+        repo = "bpftool";
+        rev = "05940344f5db18d0cb1bc1c42e628f132bc93123";
+        hash = "sha256-g2gjixfuGwVnFlqCMGLWVPbtKOSpQI+vZwIZciXFPTc=";
+        fetchSubmodules = true;
+      };
+
+      buildInputs = [ llvmPackages_16.clang llvmPackages_16.libllvm.dev elfutils zlib ];
+
+      buildPhase = ''
+        make -C src
+      '';
+
+      installPhase = ''
+        mkdir -p $out/include/bpf
+        cp libbpf/src/* $out/include/bpf
+        mkdir -p $out/bin
+        cp src/bpftool $out/bin
+      '';
+    };
+
+  vmlinux = stdenv.mkDerivation
+    {
+      pname = "vmlinux";
+      version = "eunomia-edition-20230514";
+
+      src = fetchFromGitHub {
+        owner = "eunomia-bpf";
+        repo = "vmlinux";
+        rev = "933f83becb45f5586ed5fd089e60d382aeefb409";
+        hash = "sha256-CVEmKkzdFNLKCbcbeSIoM5QjYVLQglpz6gy7+ZFPgCY=";
+      };
+
+      installPhase = ''
+        runHook preInstall
+        cp -r $src $out
+        runHook postInstall
+      '';
+    };
+
+in
+llvmPackages_16.stdenv.mkDerivation rec{
+  pname = "ecc";
+  version = "1.0.10";
+
+  src = fetchFromGitHub {
+    owner = "eunomia-bpf";
+    repo = "eunomia-bpf";
+    rev = "v${version}";
+    hash = "sha256-jqfrRpRLa37eMBX8TC4NgCjZ0RlmReI/bt9uQNxdlUs=";
+  };
+
+  cargoRoot = "compiler/cmd";
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ]
+  ++
+  (with rustPlatform;
+  [
```

Personal indentation preference (looks better in diffs):

```suggestion
  ] ++ (with rustPlatform; [
```